akka-javasdk/src/main/scala/akka/javasdk/impl/http/HttpClientImpl.scala

```diff
@@ -177,22 +176,23 @@ private[akka] final case class RequestBuilderImpl[R](
     timeout,
     request,
     { (res: HttpResponse, bytes: ByteString) =>
-      try if (res.status.isFailure) {
-        onResponseError(res, bytes)
-      } else if (res.entity.getContentType == ContentTypes.APPLICATION_JSON)
-        new StrictResponse[T](res, JsonSupport.decodeJson(`type`, bytes))
-      else if (!res.entity.getContentType.binary && (`type` eq classOf[String]))
-        new StrictResponse[T](
-          res,
-          new String(
-            bytes.toArrayUnsafe(),
-            res.entity.getContentType.getCharsetOption
-              .map[Charset]((c: HttpCharset) => c.nioCharset)
-              .orElse(StandardCharsets.UTF_8)).asInstanceOf[T])
-      else
-        throw new RuntimeException(
-          "Expected to parse the response for " + request.getUri + " to " + `type` + " but response content type is " + res.entity.getContentType)
-      catch {
+      try {
+        if (res.status.isFailure) {
+          onResponseError(res, bytes)
+        } else if (!res.entity.getContentType.binary && (`type` eq classOf[String])) {
+          new StrictResponse[T](
+            res,
+            new String(
+              bytes.toArrayUnsafe(),
+              res.entity.getContentType.getCharsetOption
+                .map[Charset]((c: HttpCharset) => c.nioCharset)
+                .orElse(StandardCharsets.UTF_8)).asInstanceOf[T])
+        } else if (res.entity.getContentType == ContentTypes.APPLICATION_JSON) {
+          new StrictResponse[T](res, JsonSupport.decodeJson(`type`, bytes))
+        } else
+          throw new RuntimeException(
+            "Expected to parse the response for " + request.getUri + " to " + `type` + " but response content type is " + res.entity.getContentType)
+      } catch {
```

Comment:
Added some `{ }` to make it easier to read. It was confusing to have a `try/catch` and `if/else` with different curly-braces usages.

But the real change is that I swapped the first `else if` with the second.
If non-binary and user asks for `String`, we return it as is without trying to decode it to String using the json decoder.